benches/bench.rs

```diff
@@ -73,9 +73,31 @@ macro_rules! generate_benches {
     };
 }
 
-fn criterion_benchmark(c: &mut Criterion) {
-    c.with_plots();
+fn benchmark_non_power_of_two<const L: usize>(b: &mut Bencher) {
+    b.iter(|| {
+        let mut rb = AllocRingBuffer::with_capacity_non_power_of_two(L);
+
+        for i in 0..1_000_000 {
+            rb.push(i)
+        }
```

Comment:
I had a look at the compiled machine code for some of these benchmarks, and for at least one of them, LLVM managed to constant-fold and unroll the entire loop to something like 4,096 `MOV`s.  That's probably not representative of common use cases - there would be enough work happening in each iteration that such degree of optimization would not be possible.  I think we should use `black_box`.